.github/workflows/reusable_promote_prerelease.yaml

```diff
@@ -0,0 +1,56 @@
+name: Promote Prerelease to Release
+
+# This workflow checks for the latest successful prerelease, excluding any labeled with "artifacts-pending" or "pre-release-failure".
+# If a suitable prerelease is found, it promotes it to a full release. This action should consequently trigger any workflows
+# that respond to the 'released' event type, ultimately invoking 'reusable_on_release.yaml'. This facilitates an automated
+# continuation of the release process.
+
+#  Usage from the caller workflow:
+#  jobs:
+#    promote_prerelease:
+#      uses: newrelic/coreint-automation/.github/workflows/reusable_promote_prerelease.yaml@v2
+#      secrets:
+#        bot_token: ${{ secrets."github_token_name" }}
+
+on:
+  workflow_call:
+    inputs:
+      block_endpoint:
+        description: This endpoint should respond status 200 for the promotion to continue.
+        required: false
+        type: string
+        default: https://newrelic.github.io/coreint-automation/automatic_release_enable
+    secrets:
+      bot_token:
+        description: "github token"
+        required: true
+
+jobs:
+  check_block_endpoint:
+    name: Check block endpoint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check block endpoint
+        run: |
+          http_status=$(curl -s -o /dev/null -w "%{http_code}" "${{ inputs.block_endpoint }}")
+
+          if [ "$http_status" != 200 ]; then
+            echo "Fail checking release block endpoint: ${{ inputs.block_endpoint }}"
+            exit 1
+          fi
+
+  promote_prerelease:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Promote previous prerelease to release
+        env:
+          GH_TOKEN: "${{ secrets.bot_token }}"
+        run: |
+          PRERELEASE_INFO=$(gh release list --repo ${{ github.repository }} --limit 1 --prerelease | grep -v "artifacts-pending" | grep -v "pre-release-failure")
```

Comment:
isn't there the risk to upgrade old pre-releases not released yet? In some repos doing so could overwrite latest

Reply:
I think we should consider a prerelease only if it does not exist a release "newer"

Reply:
I think we might list just the latest release and check if it is a prelease and if so upgrade it

Reply:
Updated in 5d68d2186e47e13cb1ac7f4b419046d66e3a410f.